compiler/rustc_mir_build/src/build/expr/as_place.rs

```diff
@@ -607,12 +630,12 @@ impl<'a, 'tcx> Builder<'a, 'tcx> {
         // The "retagging" transformation (for Stacked Borrows) relies on this.
         let idx = unpack!(block = self.as_temp(block, temp_lifetime, index, Mutability::Not,));
 
-        block = self.bounds_check(block, base_place.clone(), idx, expr_span, source_info);
+        block = self.bounds_check(block, &base_place, idx, expr_span, source_info);
 
         if is_outermost_index {
             self.read_fake_borrows(block, fake_borrow_temps, source_info)
         } else {
-            base_place = base_place.expect_upvars_resolved(self);
+            assert!(base_place.resolve_upvar_mut(self));
             self.add_fake_borrows_of_base(
                 &base_place,
```

Comment:
Should this method take a simple `Place`?

Reply:
Yeah that works.